doc: Updated dashboard documentation (features, SSL config)

LenzGr opened the pull request:
Updated Ceph manager dashboard documentation: added SSL to the feature overview, added new section about configuring SSL certificates.

Signed-off-by: Lenz Grimmer <lgrimmer@suse.com>

LenzGr commented:
jenkins render docs

ceph-jenkins commented:
Doc render available at http://docs.ceph.com/ceph-prs/21865/

rjfd commented on doc/mgr/dashboard.rst:
```diff
@@ -94,6 +97,45 @@ a list like so::
+  $ ceph config-key set mgr mgr/dashboard/key -i dashboard.key
+
+.. note::
+  You need to restart the Ceph manager processes after changing the
```
This is only true if you run the above `config-key` commands after enabling the dashboard module.
Maybe we can add that "If dashboard module is already running, then you need to reload the dashboard module ..."

LenzGr replied:
It was my understanding that reloading the dashboard module is not sufficient when using the `ceph config-key set` command, as the entire manager instance needs to be restarted in order for this change to take effect. @jcsp can you confirm?

jcsp replied:
You do need the daemon to restart, so doing a "ceph mgr fail mgr.<id>" is the natural way to do that.  However, disabling and re-enabling the module also works because ceph-mgr respawns when the list of enabled modules changes.

jcsp commented on doc/mgr/dashboard.rst:
```diff
@@ -94,6 +97,45 @@ a list like so::
+The ``dashboard.crt`` file should then be signed by your organization's CA
+(certificate authority). Once that is done, you can set it with::
+
+  $ ceph config-key set mgr mgr/dashboard/$name/crt -i dashboard.crt
```
`s/config-key set mgr mgr/config-key set mgr/`

(The extra " mgr " token was for the "config set" command only)

jcsp commented:
I'd suggest mentioning the "create-self-signed-cert" command right before "ceph mgr module enable dashboard", so that users who are following it in sequence will get a working setup

sebastian-philipp commented on doc/mgr/dashboard.rst:
```diff
@@ -94,6 +97,45 @@ a list like so::
+hostname). If all manager instances are to share the same certificate,
+you can leave off the ``$name`` portion::
+
+  $ ceph config-key set mgr mgr/dashboard/crt -i dashboard.crt
```
I'm move the option without `$name` a bit upwards. Having a single certificate for all mgr instances should be the default and thus be the first option documented.

LenzGr replied:
Good point, will change this accordingly.

LenzGr commented:
I think there is some ambiguity in what "creating a self-signed certificate" actually means. I was referring to the built-in command that generates a self-signed certificate automatically (`ceph dashboard create-self-signed-cert`).

But of course you can also generate self-signed certificates using the `openssl` command line utility.

What should be our recommended approach? Should we document it the "hard way" first, explaining how to generate and import a custom certificate?

For the sake of getting the dashboard up and running quickly, I'd be in favor of explaining how to use the built-in method for generating a certificate first, followed by instructions on how this certificate can be replaced with a proper/official one later, once the dashboard is up and running (and eventually providing a way to use the UI for replacing it).

jcsp commented:
> I'd suggest mentioning the "create-self-signed-cert" command right before "ceph mgr module enable dashboard", so that users who are following it in sequence will get a working setup
> @jcsp But doesn't the command ceph dashboard create-self-signed-cert only work after I have enabled the dashboard?

Ah, you're right of course.  So I guess just need to be very clear in the text that "mgr module enable" is not the last step, so that people don't get confused when the dashboard isn't up and running immediately.

jcsp commented:
> For the sake of getting the dashboard up and running quickly, I'd be in favor of explaining how to use the built-in method for generating a certificate first

:+1:

LenzGr commented:
jenkins render docs

LenzGr commented:
jenkins render docs

LenzGr commented:
jenkins render docs

ceph-jenkins commented:
Doc render available at http://docs.ceph.com/ceph-prs/21865/

LenzGr commented:
@jcsp and @sebastian-philipp I've reworked the structure and hopefully addressed all of your concerns: http://docs.ceph.com/ceph-prs/21865/mgr/dashboard/ - I'd be grateful for another review.

LenzGr commented:
Crap, I accidentally pushed the master branch without committing the amended changeset, which caused github to close this branch. Sorry about that.

ceph-jenkins commented:
Doc render available at http://docs.ceph.com/ceph-prs/21865/
